ui/src/components/fields/FieldMap.tsx

```
@@ -195,8 +195,15 @@ export default class FieldMap extends FieldBase {
     getDisplayValue() {
         const { prefix, values } = this.props;
         const value = getValue(values, prefix);
-        return <dl className="map">{Object.keys(value).map(key => {
-            return <div key={key}><dt>{String(key)}</dt><dd>{getValue(values, prefix + "." + key)}</dd></div>;
-        })}</dl>
+        try {
+            return <dl className="map">{Object.keys(value).map(key => {
+                return <div key={key}><dt>{String(key)}</dt><dd>{getValue(values, prefix + "." + key)}</dd></div>;
+            })}</dl>
+        }
+        catch (ex) {
+            // data is in invalid format - this occurs if data is loaded concurrently while rendering.
+            // It will fix itself during further browser refreshes
+            return null;
+        }
```

Should we explicitly check that the value is not `null`? Maybe we leave the try-catch in but log an error. But it would be good to distinguish a `null` value from one that cannot be converted to an object.

---

the `getValue()` function can return a `null`, `undefined`, `"string"`, `["array"]`, `{"object":"value"}`, `{}`, `[]`, `[null]` or similar constructs while the UI is switching contexts and renders in parallel. Flooding the console.log during a render refresh wouldn't make sense if it gets the correct value a few milliseconds later. If there is a "real" issue, I'd prefer to hide the values instead of confusing a user with error messages on specific values they might not even be interested in.

---

Okay, if it is expected that the conversion may fail due to context switching, then it is fine to suppress the error silently in all cases.